docs/deep-dives/mobile/enrollment-state-machine.mdx

```diff
@@ -0,0 +1,176 @@
+---
+id: mobile-enrollment-state-machine
+title: Enrollment State Machine
+slug: /mobile-enrollment-state-machine
+---
+
+## Possible States
+
+Within the Nimbus SDK there are multiple states, most of which have a "reason" associated with them.
+The reason field functions as a child state, and is required for parent states that have reasons.
+
+### Enrolled
+
+#### Qualified
+
+The client is not already enrolled in a recipe of the same type that would result in a feature conflict, and matched targeting and bucketing criteria for this recipe<sup><a href="#notes">[1]</a></sup>.
+
+#### Opt In
+
+The client explicitly opted into this recipe<sup><a href="#notes">[2]</a></sup>.
+
+### Not Enrolled
+
+:::info
+This state and its reasons are for enrollments that have never been enrolled, and did not enroll.
```

Comment:
>  that have never been enrolled, and did not enroll.

Wait I thought that `NotEnrolled - NotSelected` could happen if the client was enrolled in a rollout--for example at 100%--and then the rollout changed to 0%? In that case, "never been enrolled" is not true 🤔

Reply:
Oh, I see. That case would `Disqualified::NotSelected`, not `NotEnrolled::NotSelected`